src/main/java/com/gluonhq/substrate/target/LinuxTargetConfiguration.java

```diff
@@ -264,4 +268,23 @@ String getLinker() {
         }
         return "aarch64-linux-gnu-gcc";
     }
+
+    @Override
+    Predicate<Path> getTargetSpecificNativeLibsFilter() {
+        return this::checkFileArchitecture;
+    }
+
+    private boolean checkFileArchitecture(Path path) {
+        try {
+            ProcessRunner pr = new ProcessRunner("objdump", "-f", path.toFile().getAbsolutePath());
+            int op = pr.runProcess("objdump");
+            if (op == 0) return true;
+        } catch (IOException e) {
```

Comment:
Collapse catch blocks. Instead of `printStackTrace`, use log.severe `e.getMessage()`?